src/main/java/org/example/homework4/LeapYearCalculator.java

```diff
@@ -0,0 +1,19 @@
+package org.example.homework4;
+
+public class LeapYearCalculator {
+    public static void main(String[] args) {
+        int[] years = {1999, 2000, 2100, 2400, 1804, 0 };
+        for (int year : years) {
+            System.out.println(year + ": " + isLeapYear(year));
+        }
+    }
+    public static boolean isLeapYear(int year) {
+        if (year < 1 || year > 9999) {
+            return false;
+        } else if (year % 4 == 0 && year % 100 != 0) {
```

Comment:
The solution is correct, but from my point of view combining else-if and else condition together will be more clear. e.g.
```
        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            return true;
        }
```